profile-widget/qmlprofile.cpp

```diff
@@ -62,6 +71,18 @@ void QMLProfile::paint(QPainter *painter)
 	// shift the painter (taking its existing transformation into account)
 	QTransform profileTransform = QTransform();
 	profileTransform.scale(sx, sy);
+
+	// this is a bit complex because of the interaction with the Qt widget scaling
+	// logic. We use the default setup where the scale happens with the center of the
+	// widget as transformation center.
+	// When zooming and panning we want to ensure that we always show a subset of the
+	// profile and not the "empty space" around the profile.
+	// a bit of math on a piece of paper shos that our offsets need to stay within +/- dx and dy
+	qreal dx = width() * (scale() - 1) / (2 * dpr * scale());
```

Comment:
I reckon that `scale` is limited to `>=1` (i.e. only zoom in, no zoom out), otherwise `dx` and `dy` become negative and the clamping below will become funny.

Reply:
Correct, but that's done in QML, so it might make sense to enforce that here as well. I'll think of a good solution for that.